### PR TITLE
Implement a new exterminate action

### PR DESCRIPTION
**Pull request by frankban**
The previous action no longer worked because the charm now installs dependencies in a virtalenv.
The new exterminate action allows for more observability of results, and also for selecting specific containers.

### CODE REVIEW

File: lib/charms/layer/jujushell.py [hunk 1]
```diff
@@ -323,13 +323,31 @@ def setup_lxd():
...
-        call('/snap/bin/lxc', 'delete', '-f', container.name)
+        if name and (container.name != name):
+            continue
+        is_running = container.status.lower() == 'running'
+        if only_stopped and is_running:
+            continue
+        removed.append(container.name)
+        if dry:
+            continue
+        if is_running:
```
Comment (jcsackett) [inline lib/charms/layer/jujushell.py:347]: Really nice flow control with `continue` there.

File: tests/test_jujushell.py [hunk 2]
```diff
@@ -725,6 +725,187 @@ def test_initialized(self, mock_log):
...
+                'delete': Mock(),
+            }) for name, running in containers
+        ]
+        return patch('jujushell._lxd_client', type('Client', (object, ), {
+            'containers': type('Containers', (object,), {
+                'all': lambda: results,
+            }),
+        }))
+
+
```
Comment (jcsackett) [inline tests/test_jujushell.py:908]: Thanks for the tests!

File: actions/exterminate [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Canonical Ltd.
+# Licensed under the AGPLv3, see LICENCE file for details.
+
+# Load modules from $JUJU_CHARM_DIR/lib.
+import sys
+sys.path.append('lib')
```
Comment (jcsackett) [inline actions/exterminate:8]: Is this going to work? This is a relative path, so I guess it's relying on juju executing the action from the correct directory?

**Review by jcsackett — APPROVED** [review]

File: actions/exterminate (hunk 3 above)
Comment (frankban) [reply actions/exterminate:8]: Yes, it's relying on that.

**Pull request merged**